src/pages/RoomInvitePage.js

```diff
@@ -180,64 +97,46 @@ function RoomInvitePage(props) {
         Navigation.navigate(backRoute);
     }, [selectedOptions, backRoute, props.report.reportID, validate]);
 
-    const headerMessage = useMemo(() => {
-        const searchValue = searchTerm.trim().toLowerCase();
-        if (!userToInvite && CONST.EXPENSIFY_EMAILS.includes(searchValue)) {
-            return translate('messages.errorMessageInvalidEmail');
-        }
-        if (!userToInvite && excludedUsers.includes(OptionsListUtils.addSMSDomainIfPhoneNumber(searchValue).toLowerCase())) {
-            return translate('messages.userIsAlreadyMember', {login: searchValue, name: reportName});
-        }
-        return OptionsListUtils.getHeaderMessage(personalDetails.length !== 0, Boolean(userToInvite), searchValue);
-    }, [excludedUsers, translate, searchTerm, userToInvite, personalDetails, reportName]);
     return (
         <ScreenWrapper
             shouldEnableMaxHeight
             testID={RoomInvitePage.displayName}
         >
-            {({didScreenTransitionEnd}) => {
-                const sections = didScreenTransitionEnd ? getSections() : [];
-
-                return (
-                    <FullPageNotFoundView
-                        shouldShow={_.isEmpty(props.report)}
-                        subtitleKey={_.isEmpty(props.report) ? undefined : 'roomMembersPage.notAuthorized'}
-                        onBackButtonPress={() => Navigation.goBack(backRoute)}
-                    >
-                        <HeaderWithBackButton
-                            title={translate('workspace.invite.invitePeople')}
-                            subtitle={reportName}
-                            onBackButtonPress={() => {
-                                Navigation.goBack(backRoute);
-                            }}
+            {({didScreenTransitionEnd}) => (
+                <FullPageNotFoundView
+                    shouldShow={_.isEmpty(props.report)}
+                    subtitleKey={_.isEmpty(props.report) ? undefined : 'roomMembersPage.notAuthorized'}
+                    onBackButtonPress={() => Navigation.goBack(backRoute)}
+                >
+                    <HeaderWithBackButton
+                        title={translate('workspace.invite.invitePeople')}
+                        subtitle={reportName}
+                        onBackButtonPress={() => {
+                            Navigation.goBack(backRoute);
+                        }}
+                    />
+                    <MemberInviteList
+                        didScreenTransitionEnd={didScreenTransitionEnd}
+                        inviteUsers={inviteUsers}
+                        selectedOptions={selectedOptions}
+                        setSelectedOptions={setSelectedOptions}
```

Comment:
I feel like `RoomInvitePage` and `WorkspaceInvitePage` has no business in knowing about `selectedOptions`. Ideally it would be an internal state (and for consistency we should probably rename it to `selectedMembers` IMO) to `MemberInviteList` and `inviteUsers` callback would get it as a parameter. The only issue here is the fact that the `FormAlertWithSubmitButton` bellow need to know whether we have a selected user or not so it can probably enable/disable. I see a couple ways to deal with this:

* We could move `FormAlertWithSubmitButton` inside `MemberInviteList`;
* We get rid of `selectedOptions` as this would be managed internally but we keep `setSelectedOptions` to give the parent context about whether we have something selected or not (we would not need to save the whole `selectedOptions`, we just set another state like `hasSelectedMembers = selectedOptions.length > 0` and use it later. Or we rename `setSelectedOptions` to something like `setHasSelectedMembers` and handle this logic internally to `MemberInviteList`.

I kinda like the idea of moving the submit button inside the member invite list (we'd need a few more props passed down to it) but at the same time I wonder if it's really a good idea because now the parent does not have the flexibility of setting its own UI as for how to render this button. Might not really be a big deal because we want consistency and doing this would force consistency across all pages that use it.

What are your thoughts on this?

Reply:
@tienifr forgot to tag you in the comment above